src/VirtualClient/VirtualClient.Actions/Sysbench/SysbenchClientExecutor.cs

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Polly;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Contracts;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+    using VirtualClient.Contracts.Metadata;
+
+    /// <summary>
+    /// The Sysbench Client workload executor.
+    /// </summary>
+    public class SysbenchClientExecutor : SysbenchExecutor
+    {
+        private static readonly string[] SingleTableWorkloads =
+        {
+            "select_random_points",
+            "select_random_ranges"
+        };
+
+        private string sysbenchExecutionArguments;
+        private string sysbenchLoggingArguments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SysbenchClientExecutor"/> class.
+        /// </summary>
+        /// <param name="dependencies">An enumeration of dependencies that can be used for dependency injection.</param>
+        /// <param name="parameters">An enumeration of key-value pairs that can control the execution of the component.</param>
+        public SysbenchClientExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters = null)
+            : base(dependencies, parameters)
+        {
+            this.ClientFlowRetryPolicy = Policy.Handle<Exception>().RetryAsync(3);
+            this.PollingTimeout = TimeSpan.FromMinutes(40);
+        }
+
+        /// <summary>
+        /// The total time of execution option passed to Sysbench.
+        /// </summary>
+        public TimeSpan Duration
+        {
+            get
+            { 
+                return this.Parameters.GetTimeSpanValue(nameof(SysbenchClientExecutor.Duration), TimeSpan.FromMinutes(5));
+            }
+        }
+
+        /// <summary>
+        /// The workload option passed to Sysbench.
+        /// </summary>
+        public string Workload
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SysbenchClientExecutor.Workload));
+            }
+        }
+
+        /// <summary>
+        /// The workload option passed to Sysbench.
+        /// </summary>
+        public int TableCount
```

Comment:
Code Implementation Recommendation: Avoid significant logic in C# property getters and setters.

Move the logic here for defining the TableCount to the InitializeAsync() method to be consistent with the relative convention in other VC components.

In fact, given we are using similar/same logic in multiple components (e.g. SysbenchClientExecutor, SysbenchConfiguration, SysbenchExecutor), I recommend creating a static properties and methods on the SysbenchConfiguration class to handle this logic.

e.g.
public const int DefaultTableCount = 10;

public static int GetTableCount(ISystemManagement systemManagement, string databaseScenario);
public static int GetRecordCount(ISystemManagement systemManagement, string databaseScenario);

...then use these statics in each of the individual components (i.e. in the InitializeAsync method) to set the TableCount and RecordCount properties to values (if they are not already set).